gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/AbstractPluginEventListener.java

```diff
@@ -127,17 +126,39 @@ private void deployPlugin(Plugin plugin, Map<String, List<Plugin>> resolvedDepen
 
     private record PluginKey(String id, String type) {}
 
-    private static class PluginComparator implements Comparator<Plugin> {
+    protected static class PluginComparator implements Comparator<Plugin> {
+
+        private final String loadFirst;
+
+        public PluginComparator(String loadFirst) {
+            this.loadFirst = loadFirst;
+        }
 
         @Override
         public int compare(Plugin o1, Plugin o2) {
-            Integer pos1 = pluginPriority.indexOf(o1.type());
-            Integer pos2 = pluginPriority.indexOf(o2.type());
+            int pos1 = pluginPriority.indexOf(o1.type());
+            int pos2 = pluginPriority.indexOf(o2.type());
 
             if (pos1 >= 0) {
                 if (pos2 >= 0) {
+                    if (pos1 == pos2) {
+                        if (
+                            loadFirst != null &&
+                            o1.manifest().type().equalsIgnoreCase(SECRET_PROVIDER) &&
```

Comment:
You probably want to test the plugin type first, then make a condition on `loadFirst`. If it is not a secret provider plugin then `loadFirst` won't matter, value is bound to this specific type of plugin.

```pseudo code
if pluginType == "secret provider" and loadFirst is not null or empty 
  if o1.id == loadFirst
   return -1
  else if o2.id == loadFirst
  return 1
end if
return comparison of positions
```

This way you respect the contract and make it clear why you compare `loadFirst`

But let's wait for the above comment to be answered

Reply:
This has then to be changed